crates/sui-types/src/temporary_store.rs

```diff
@@ -1008,14 +1000,7 @@ impl<S: ObjectStore> TemporaryStore<S> {
             old_obj.storage_rebate
         } else {
             // else, this is a dynamic field, not an input object
-            if let Ok(Some(old_obj)) = self.store.get_object(id) {
-                if old_obj.version() != expected_version {
-                    // not a lot we can do safely and under this condition everything is broken
-                    panic!(
-                        "Expected to find old object with version {expected_version}, found {}",
-                        old_obj.version(),
-                    );
-                }
+            if let Ok(Some(old_obj)) = self.store.get_object_by_key(id, expected_version) {
```

Comment:
panic / expect if None?

Reply:
isn't that what is happening? this is panicking on `None`.
I have gated the code so that previous version will not change behavior (panic on version mismatch), whereas this is just looking for a specific version and panicking if `None`